src/httpserver.cpp

```diff
@@ -426,6 +520,26 @@ bool InitHTTPServer()
     int workQueueDepth = std::max((long)gArgs.GetArg("-rpcworkqueue", DEFAULT_HTTP_WORKQUEUE), 1L);
     LogPrintf("HTTP: creating work queue of depth %d\n", workQueueDepth);
 
+    std::vector<evconnlistener*> listeners;
+    for (const auto& bind_handle : boundSockets) {
+        evconnlistener* listener = evhttp_bound_socket_get_listener(bind_handle);
+        evutil_socket_t sock = evhttp_bound_socket_get_fd(bind_handle);
+        SetListenSocketDeferred(sock);
+        listeners.push_back(listener);
+    }
+    g_limiter = MakeUnique<ConnectionLimiter>(std::move(listeners), workQueueDepth * 2);
+    evhttp_set_gencb(http, http_request_cb, g_limiter.get());
+
+#if LIBEVENT_VERSION_NUMBER >= 0x02020001
```

Comment:
Would like to see this defined separately as:

```c++
#if LIBEVENT_VERSION_NUMBER >= 0x02020001
#define HAVE_EVHTTP_SET_NEWREQCB 1
#endif
```

That way the check here is more obvious, particularly since you do the check elsewhere. Of course, even better than this would be to use `AC_CHECK_FUNC()`...